tests/src/LuaTests.cs

```diff
@@ -2795,6 +2795,32 @@ public void TestGuid()
                 Assert.AreEqual(new Guid("adc70ae1-769e-4ace-aa83-928a604c5739"), o);
             }
         }
+        
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        public void TestMaximumRecursion(int maxRecursion)
+        {
+            var tc = new TestClass();
+            tc.LongValue = 5;
+            
+            using (Lua lua = new Lua())
+            {
+                lua.MaximumRecursion = maxRecursion;
+                lua.LoadCLRPackage();
+                lua["myTc"] = tc;
+                
+                if(maxRecursion == 0)
+                    Assert.AreEqual(1,lua.Globals.Count()); //register only the root reference
+                else
+                    Assert.Greater(lua.Globals.Count(),1); //many globals registered (all sub properties)
```

Comment:
NUnitLite doesn't have `Greater` you can just do `True (lua.Globals.Count(),1)`

Reply:
Thanks for helping me with this PR.  Looks like there are some failing tests with `PlatformNotSupportedException` in NLua.CodeGeneration constructor now...